test/Quake/arg_subst_func.qke

```diff
@@ -146,3 +146,33 @@ func.func @testy4(%arg0: !cc.stdvec<!cc.struct<{i32, f64, i8, i16}>>) {
 // CHECK:           call @callee4(%[[VAL_32]]) : (!cc.stdvec<!cc.struct<{i32, f64, i8, i16}>>) -> ()
 // CHECK:           return
 // CHECK:         }
+
+func.func @testy5(%arg0: !cc.ptr<!cc.state>) {
+  %0 = quake.get_number_of_qubits %arg0 : (!cc.ptr<!cc.state>) -> i64
+  %1 = quake.alloca !quake.veq<?>[%0 : i64]
+  %5 = quake.init_state %1, %arg0 : (!quake.veq<?>, !cc.ptr<!cc.state>) -> !quake.veq<?>
+  return
+}
+
+func.func @init(%arg0: i64, %arg1: !quake.veq<?>) -> !quake.veq<?> {
+  return %arg1 : !quake.veq<?>
+}
+
+func.func @num_qubits(%arg0: i64) -> i64 {
```

Comment:
When we get to the phase of building the substitutions, we should know the number of qubits as a constant, right?